keycloak-event-emitter/src/main/java/io/cloudtrust/keycloak/eventemitter/EventEmitterProvider.java

```diff
@@ -215,4 +225,18 @@ private void send(HttpPost httpPost) throws EventEmitterException, IOException {
         }
     }
 
+    private void completeEventAttributes(Event event) {
+        // add missing username
+        if (event.getDetails() == null) {
+            event.setDetails(new HashMap<>());
+        }
+        String username = event.getDetails().get(Details.USERNAME);
+        if (!Strings.isNullOrEmpty(event.getUserId()) && Strings.isNullOrEmpty(username)) {
+            // retrieve username from userId
+            UserModel user = keycloakSession.users().getUserById(event.getUserId(),
```

Comment:
It was impossible to retrieve the username of the current user from the session/context ?

Reply:
I'm afraid it is not possible, since we do not have any context